R/limits.r

```diff
@@ -184,10 +184,9 @@ limits.POSIXlt <- function(lims, var, call = caller_env()) {
 expand_limits <- function(...) {
   data <- list2(...)
   data_dfs <- vapply(data, is.data.frame, logical(1))
-  data <- do.call(c, c(list(data[!data_dfs]), data[data_dfs]))
+  data <- unlist(c(list(data[!data_dfs]), data[data_dfs]), recursive = FALSE)
   n_rows <- max(vapply(data, length, integer(1)))
-  data <- lapply(data, rep, length.out = n_rows)
-  data <- new_data_frame(data)
+  data <- data_frame(!!!lapply(data, rep, length.out = n_rows), .name_repair = "minimal")
```

Comment:
I feel like we get better tracebacks if we avoid doing complex computations in a `!!!`

Reply:
I guess this function doesn't do standard tidyverse recycling? i.e. this would be an error if it did

``` r
ggplot2:::expand_limits(x = 1:2, y = 1:4)$data
#>   x y
#> 1 1 1
#> 2 2 2
#> 3 1 3
#> 4 2 4
```

<sup>Created on 2022-06-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
@DavisVaughan this is not intended to be seen as data construction in any way - just a way to specify multiple limits for different aesthetics at the same time. Having it adhere to any recycling rules would not make much sense